R/naive_direction_forecaster.R

```diff
@@ -0,0 +1,163 @@
+      as_of = strftime(forecast_due_date, "%Y-%m-%d")
+    ) %>% as_tibble()
+  )
+
+  # By the cadence of HHS data, we expect to have data up to Friday 12 days back
+  # from the forecast due date. We get the latest 7 day sum of the data available.
+  reference_7d_counts <- short_snapshot %>%
+    filter(time_value <= forecast_due_date - 12L) %>%
+    group_by(geo_value) %>%
+    complete(time_value = full_seq(time_value, 1L)) %>%
+    slice_max(time_value, n = 7L) %>%
+    summarize(
+      reference_7dcount = sum(value),
+      time_value = max(time_value),
+      .groups = "drop"
+    )
+
+  # Rule of three for probability of "novel behavior"; mixing weight for uniform
+  uniform_forecaster_weight <- 3 / (
+    # roughly, how many years of hhs influenza are available
+    as.numeric(forecast_due_date - as.Date("2020-10-16")) / (365 + 1 / 4 - 1 / 100 + 1 / 400) *
+      # say there's one wave/season per year
+      1 *
+      # say states from separate HHS Regions are "separate" enough to count as different data points, while those within the same region aren't separate enough; 10 HHS Regions = 10 buckets
+      10
+  )
+
+  direction_predictions <- quantile_predictions %>%
+    group_by(
+      forecaster,
+      data_source,
+      signal,
+      geo_value,
+      incidence_period,
+      reference_date,
+      target,
+      horizon,
+      target_end_date,
+      location
+    ) %>%
+    summarize(
+      forecast_acdf = list(approx_cdf_from_quantiles(value, output_type_id)),
+      .groups = "keep"
+    ) %>%
+    left_join(augmented_location_data %>% select(-geo_value), by = "location") %>%
+    left_join(reference_7d_counts, by = "geo_value") %>%
+    reframe(
+      output_type = "pmf",
+      output_type_id = c("large_decrease", "decrease", "stable", "increase", "large_increase"),
+      value = {
+        stopifnot(length(forecast_acdf) == 1L)
+        if (horizon == -1) {
+          thresh <- increase_count_1_thresh
+        } else if (horizon == 0) {
+          thresh <- increase_count_2_thresh
+        } else if (horizon == 1) {
+          thresh <- increase_count_3_thresh
+        } else if (horizon == 2) {
+          thresh <- increase_count_4_thresh
+        } else if (horizon == 3) {
+          thresh <- increase_count_5_thresh
+        } else {
+          stop("horizon must be in -1:3")
+        }
+        p_large_dec <- p_le(forecast_acdf[[1L]], reference_7dcount - 2 * thresh)
+        p_large_or_nonlarge_dec <- p_le(forecast_acdf[[1L]], reference_7dcount - thresh)
+        p_large_or_nonlarge_inc <- p_ge(forecast_acdf[[1L]], reference_7dcount + thresh)
+        p_large_inc <- p_ge(forecast_acdf[[1L]], reference_7dcount + 2 * thresh)
+        result <- c(
+          p_large_dec,
+          p_large_or_nonlarge_dec - p_large_dec,
+          1 - p_large_or_nonlarge_dec - p_large_or_nonlarge_inc,
+          p_large_or_nonlarge_inc - p_large_inc,
+          p_large_inc
+        )
+        result
+      }
```

Comment:
todo: also maybe need to rename the targets here to the categorical trend target names?

Since there are separate `target` and `ahead` columns maybe you could just `mutate` to change `target` here.

------

(earlier rambling when I thought target included the ahead)

Before `reframe` you may need an `%>% ungroup(target)` so you can output its new value.  ... or not, at least in my current version...
```r
tibble(g=c(1,1,1,2,3), v=1:5) %>% group_by(g) %>% reframe(g=paste0(g,"p"), sum(v))
#> # A tibble: 5 × 2
#>   g     `sum(v)`
#>   <chr>    <int>
#> 1 1p           6
#> 2 1p           6
#> 3 1p           6
#> 4 2p           4
#> 5 3p           5
```
 but IIRC `summarize` took issue in some version so it might be safer do the `ungroup(target)` thing beforehand, idk.